Apps/UnitTests/Shared/Tests.h

```diff
@@ -98,6 +98,82 @@ TEST(NativeAPI, LifeCycle)
     }
 }
 */
+
+TEST(Performance, Spheres)
+{
+    // create a bunch of sphere, does the rendering for a number of frames, log time it took
+    static const std::string script{ R"(
+        console.log("Setting up Performance test.");
+        var engine = new BABYLON.NativeEngine();
+        var scene = new BABYLON.Scene(engine);
+
+        var size = 12;
+        for (var i = 0; i < size; i++) {
+            for (var j = 0; j < size; j++) {
+                for (var k = 0; k < size; k++) {
+                    var sphere = BABYLON.Mesh.CreateSphere("sphere" + i + j + k, 32, 0.9, scene);
+                    sphere.position.x = i;
+                    sphere.position.y = j;
+                    sphere.position.z = k;
+                }
+            }
+        }
+
+        scene.createDefaultCamera(true, true, true);
+        scene.activeCamera.alpha += Math.PI;
+        scene.createDefaultLight(true);
+        engine.runRenderLoop(function () {
+            scene.render();
+        });
+        console.log("Ready!");
+        SetReady();
+        )" };
+
+    Babylon::Graphics::Device device = deviceTestConfig;
+    std::optional<Babylon::Graphics::DeviceUpdate> update{};
+    std::promise<int32_t> ready;
+    update.emplace(device.GetUpdate("update"));
+
+    Babylon::AppRuntime runtime{};
+    runtime.Dispatch([&ready, &device](Napi::Env env) {
+        device.AddToJavaScript(env);
+
+        Babylon::Polyfills::Console::Initialize(env, [](const char* message, auto) {
+            printf("%s", message);
+            fflush(stdout);
+            });
+        Babylon::Polyfills::Window::Initialize(env);
+        Babylon::Plugins::NativeEngine::Initialize(env);
+        env.Global().Set("SetReady", Napi::Function::New(env, [&ready](const Napi::CallbackInfo& info)
```

Comment:
nit: use JS conventions

 ```suggestion
        env.Global().Set("setReady", Napi::Function::New(env, [&ready](const Napi::CallbackInfo& info)
```